packaging/specifiers.py

```diff
@@ -98,15 +128,18 @@ def __repr__(self):
         return "<{0}({1!r}{2})>".format(self.__class__.__name__, str(self), pre)
 
     def __str__(self):
+        # type: () -> str
         return "{0}{1}".format(*self._spec)
 
     def __hash__(self):
+        # type: () -> int
         return hash(self._spec)
 
     def __eq__(self, other):
+        # type: (object) -> bool
         if isinstance(other, string_types):
             try:
-                other = self.__class__(other)
+                other = self.__class__(str(other))
```

Comment:
Nope:

```diff
diff --git a/packaging/_compat.py b/packaging/_compat.py
index a145f7e..bbfcb16 100644
--- a/packaging/_compat.py
+++ b/packaging/_compat.py
@@ -17,7 +17,7 @@ PY3 = sys.version_info[0] == 3
 # flake8: noqa

 if PY3:
-    string_types = (str,)
+    string_types = (str,)  # mypy: ignore
 else:
     string_types = (basestring,)

diff --git a/packaging/specifiers.py b/packaging/specifiers.py
index 091e9de..5e1314e 100644
--- a/packaging/specifiers.py
+++ b/packaging/specifiers.py
@@ -139,7 +139,7 @@ class _IndividualSpecifier(BaseSpecifier):
         # type: (object) -> bool
         if isinstance(other, string_types):
             try:
-                other = self.__class__(str(other))
+                other = self.__class__(other)
             except InvalidSpecifier:
                 return NotImplemented
         elif not isinstance(other, self.__class__):
@@ -151,7 +151,7 @@ class _IndividualSpecifier(BaseSpecifier):
         # type: (object) -> bool
         if isinstance(other, string_types):
             try:
-                other = self.__class__(str(other))
+                other = self.__class__(other)
             except InvalidSpecifier:
                 return NotImplemented
         elif not isinstance(other, self.__class__):
```

```
$ tox -e mypy
GLOB sdist-make: /Users/dustiningram/git/pypa/packaging/setup.py
mypy inst-nodeps: /Users/dustiningram/git/pypa/packaging/.tox/.tmp/package/1/packaging-20.0.dev0.zip
mypy installed: mypy==0.720,mypy-extensions==0.4.1,packaging==20.0.dev0,pyparsing==2.4.2,six==1.12.0,typed-ast==1.4.0,typing-extensions==3.7.4
mypy run-test-pre: PYTHONHASHSEED='1543042496'
mypy run-test: commands[0] | mypy packaging --strict
mypy run-test: commands[1] | mypy packaging --py2 --strict
packaging/specifiers.py:142: error: Argument 1 to "_IndividualSpecifier" has incompatible type "basestring"; expected "str"
packaging/specifiers.py:154: error: Argument 1 to "_IndividualSpecifier" has incompatible type "basestring"; expected "str"
ERROR: InvocationError for command /Users/dustiningram/git/pypa/packaging/.tox/mypy/bin/mypy packaging --py2 --strict (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   mypy: commands failed
```
Open to other suggestions here, but I also don't see how this would cause problems.